lib/roby/app/scripts/test.rb

```diff
@@ -126,6 +140,20 @@
                 end
                 exit 0
             end
+        elsif force_discovery
+            test_files += discover_test_files(all: all, only_self: only_self, base_dir: base_dir)
+
+            test_files = test_files.flat_map do |file|
+                next file unless file =~ /ROBOT/
+
+                robot_type = Roby.app.robot_type
+                robot_name = Roby.app.robot_name
+                (replacements = [robot_type]) << robot_name if robot_type != robot_name
```

Comment:
would be nice having a method `Roby::Application#robots`:

```rb
def robots
  [robot_type] << robot_name if robot_type != robot_name
end
```

Reply:
It would only return an array with `[robot_type, robot_name]`? You see that the way if did and are suggesting for this function this would return nil when `robot_type == robot_name`, is that intended?

Reply:
indeed, it should be

```rb
def robots
  r = [robot_type]
  r << robot_name if robot_type != robot_name
  r
end
```

Reply:
or
```rb
def robots
  r = (robot_type != robot_name ? [robot_type] : [robot_type, robot_name])
end
```

Reply:
I would prefer `robot_configuration_names` to be more explicit. `robots` is a bit confusing.

Reply:
just created it